### PR TITLE
Move is_restart from qmc_common to QMCDriver

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Following a comment in #4127, this removes the `is_restart` member variable from `qmc_common` to `QMCDriver`. I also changed `qmc_common` to use in-class initialization and reordered members to be in decreasing order of size.

It's only used to output a line to the screen. Do we want to get rid of it entirely? 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted

### CODE REVIEW

**Comment by prckent** [comment]
I think we should remove this.

There are a bunch of old / unused parameters in the driver layers up to qmcapp. Sometimes there is also a bit of remnant plumbing as well. I suggest asking on our Slack for any questionable cases, but generally it will be much easier and cleaner to put in something new focused on the batched code than attempt to keep this functionality. This is particularly the case for anything that relies on legacy XML "parsing" in my opinion.

File: src/QMCDrivers/QMCDriver.cpp [hunk 1]
```diff
@@ -514,7 +511,7 @@ bool QMCDriver::putQMCInfo(xmlNodePtr cur)
     CurrentStep = 0;
 
   //if walkers are initialized via <mcwalkerset/>, use the existing one
-  if (qmc_common.qmc_counter || qmc_common.is_restart)
+  if (qmc_common.qmc_counter)
```
Comment (quantumsteve) [inline src/QMCDrivers/QMCDriver.cpp:514]: I'm concerned that removing `is_restart` may have unintended consequences.

**Comment by quantumsteve** [comment]
> I think we should remove this variable.

For discussion, I added an additional commit removing the `is_restart` member variable from `QMCDriver`.

**Review by ye-luo — APPROVED** [review]
Agreed with the deletion of is_restart. Rebuild is more straightforward if we really need it.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**